windows/src/global/delphi/general/Keyman.System.CanonicalLanguageCodeUtils.pas

```diff
@@ -37,6 +37,10 @@ class function TCanonicalLanguageCodeUtils.FindBestTag(const Tag: string; AddReg
     if t.Tag = '' then
       Exit('');
 
+    // Special case for IPA keyboards; otherwise we'd have und-Zyyy-fonipa
+    if (t.Language = 'und') and (t.Variant = 'fonipa') then
+      Exit('und-fonipa');
```

Comment:
What about adding this to the fonipa test in CanonicalLanguageCodeUtilsTest.pas ?